### DISCUSSION TITLE
Set Height/Width and postion of webp animation

### DISCUSSION THREAD

**Discussion by defunky** [dlemstra/Magick.NET#1160] [Help]
I'm trying to open webp animation onto a specific canvas size and position so I can compose image ontop of it. The animation is smaller than the canvas that I need so I'm trying to enlarge the canvas and move the images into the correct position (imagine placing a 100x100 animation onto 400x400 canvas and moving it into middle of it). I've tried passing in ReadSettings into the collection and defining width/height and page geometry but it didn't seem to do anything. How can I achieve this?

 I'm currently using the code below. 
```
using (var images = new MagickImageCollection("animation.webp"))
{
    images.Coalesce();

    using (var overlay= new MagickImage("overlay.png"))
    {
        foreach (var image in images)
        {
            image.Composite(overlay, CompositeOperator.Over);
        }
    }

    images.Write("output.webp");
}
```

**Comment by dlemstra** [comment]
Something like this will probably work:
```C#
foreach (var image in images)
{
    image.Extent(overlay.Width, overlay.Height, Gravity.Center);
    image.Composite(overlay, CompositeOperator.Over);
}
```
